client/src/observatory/benchmark.tsx

```diff
@@ -0,0 +1,263 @@
+import { GradeDistribution, ObservatoryResult } from "./types";
+import useSWRImmutable from "swr/immutable";
+import { OBSERVATORY_API_URL } from "../env";
+import { Loading } from "../ui/atoms/loading";
+import NoteCard from "../ui/molecules/notecards";
+import { fixMinusSymbol } from "./utils";
+
+export function useGradeDistribution(grade: string | null | undefined) {
+  return useSWRImmutable(`gradeDistribution-${grade}`, async (key) => {
+    const url = new URL(OBSERVATORY_API_URL + "/api/v2/grade_distribution");
+    const res = await fetch(url);
+    return await handleGradeDistributionResponse(res);
+  });
+}
+
+export async function handleGradeDistributionResponse(
+  res: Response
+): Promise<GradeDistribution[]> {
+  if (!res.ok) {
+    let message = `${res.status}: ${res.statusText}`;
+    try {
+      const data = await res.json();
+      if (data.error) {
+        message = data.message;
+      }
+    } finally {
+      throw Error(message);
+    }
+  }
+  return await res.json();
+}
+
+function niceNumber(range: number, round: boolean): number {
```

Comment:
Renamed and documented the function.

Since this is so tightly coupled to `benchmark.tsx`, I would prefer to leave it here. We should move it when another component starts using this.